scripts/run_multiuser_tests.py

```diff
@@ -0,0 +1,303 @@
+    ]
+
+    try:
+        print(" ".join(add_developers_cmd))
+        subprocess.check_call(add_developers_cmd)
+        print(" ".join(add_users_cmd))
+        subprocess.check_call(add_users_cmd)
+        print(" ".join(add_categories_cmd))
+        subprocess.check_call(add_categories_cmd)
+        print(" ".join(remove_categories_cmd))
+        subprocess.check_call(remove_categories_cmd)
+        print(" ".join(add_tags_cmd))
+        subprocess.check_call(add_tags_cmd)
+        print(" ".join(remove_tags_cmd))
+        subprocess.check_call(remove_tags_cmd)
+        print(" ".join(update_metadata_cmd))
+        subprocess.check_call(update_metadata_cmd)
+    except subprocess.CalledProcessError as cpe:
+        print("Error during developer actions on {}\n stdout: {}\n stderr: {}".format(
+            global_workflow_name,
+            cpe.stdout,
+            cpe.stderr
+        ))
+        raise
+
+    # As Bob, run global workflow
+    login_bob()
+
+    exec_objs = util.run_executable(
+        oid=global_workflow_name,
+        project=PROJECT,
+        test_folder=specific_applet_folder(tname),
+        test_name=tname
+    )
+
+    successes, failures = wait_for_completion(exec_objs)
+    if len(successes) > 0:
+        report_test_success(tname)
+    elif len(failures) > 0:
+        report_test_failure(tname)
+        raise RuntimeError("Analysis failed in test {}".format(tname))
+
+def main():
+    argparser = argparse.ArgumentParser(
+        description="Run dxCompiler multi-user tests on the platform"
+    )
+    argparser.add_argument(
+        "--alice-token",
+        help="Token for user-dnanexus_apps_robot on staging",
+        required=True,
+    )
+    argparser.add_argument(
+        "--bob-token",
+        help="Token for user-dnanexus_apps_test_robot on staging",
+        required=True,
+    )
+    argparser.add_argument(
+        "--folder", help="Use an existing folder, instead of building dxCompiler"
+    )
+    argparser.add_argument(
+        "--project", help="DNAnexus project ID", default="dxCompiler_playground"
+    )
+    argparser.add_argument(
+        "--clean",
+        help="Remove build directory in the project after running tests",
+        action="store_true",
+        default=False,
+    )
+    args = argparser.parse_args()
+
+    register_tokens(args.alice_token, args.bob_token)
+
+    version_id = util.get_version_id(TOP_DIR)
+    register_version(version_id)
+
+    project = util.get_project(args.project)
+    if project is None:
+        raise RuntimeError("Could not find project {}".format(args.project))
+    print("Project {} ({})".format(project.name, project.get_id()))
+    register_project(project)
+
+    # Do folder setup as Alice
+    login_alice()
+
+    if args.folder is None:
+        base_folder = util.create_build_dirs(project, version_id)
+    else:
+        # Use existing prebuilt base folder
+        base_folder = args.folder
```

Comment:
I edited the documentation for the folder arg and the comment, in both test scripts, to "Use an existing folder with dxCompiler assets, instead of building dxCompiler."

Reply:
I don't think this can be used with another version, since version_id is used in the method to compile workflows and that would probably not work when it doesn't match. It's probably just to avoid compiling the same code again (but for GHA, we always have to do a fresh compilation on the worker).